app/src/main/java/com/duckduckgo/app/global/shortcut/AppShortcutCreator.kt

```diff
@@ -23,17 +23,52 @@ import android.content.pm.ShortcutInfo
 import android.content.pm.ShortcutManager
 import android.os.Build
 import androidx.annotation.RequiresApi
+import androidx.annotation.UiThread
 import androidx.core.content.pm.ShortcutInfoCompat
 import androidx.core.graphics.drawable.IconCompat
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleObserver
+import androidx.lifecycle.OnLifecycleEvent
 import com.duckduckgo.app.bookmarks.ui.BookmarksActivity
 import com.duckduckgo.app.browser.BrowserActivity
 import com.duckduckgo.app.browser.R
+import com.duckduckgo.di.scopes.AppObjectGraph
+import com.squareup.anvil.annotations.ContributesTo
+import dagger.Module
+import dagger.Provides
+import dagger.multibindings.IntoSet
+import timber.log.Timber
 import javax.inject.Inject
+import javax.inject.Singleton
 
-class AppShortcutCreator @Inject constructor() {
+@Module
+@ContributesTo(AppObjectGraph::class)
+class AppShortcutCreatorModule {
+    @Provides
+    @IntoSet
+    fun provideAppShortcutCreatorObserver(appShortcutCreator: AppShortcutCreator): LifecycleObserver {
+        return AppShortcutCreatorLifecycleObserver(appShortcutCreator)
+    }
+}
+
+class AppShortcutCreatorLifecycleObserver(
+    private val appShortcutCreator: AppShortcutCreator
+) : LifecycleObserver {
+    @UiThread
+    @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
+    fun configureAppShortcuts() {
+        Timber.i("Configure app shortcuts")
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N_MR1) {
+            appShortcutCreator.configureAppShortcuts()
+        }
+    }
+}
```

Comment:
This class is created because `AppShortcutCreator` accesses types that are not available in all APIs, ie. `ShortcutInfo`.
Android would throw with an error message saying `Lifecycles cannot access even other methods so you should make sure that your observer classes only access framework classes that are available in your min API level OR use lifecycle:compiler annotation processor.`
Rather than adding the suggested new annotation processor, we just create a new class, `AppShortcutCreatorLifecycleObserver`, that is the lifecycle observer, that does not access any non supported method.
This is basically what the suggested annotation processor does, creating a Proxy class.